src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -878,22 +908,127 @@ namespace
     REWRITER(NGConvolutionOp)
     {
         auto convolOp = cast<NGConvolutionOp>(op);
-        auto loc = convolOp.getLoc();
-        ScopedContext scope(rewriter, loc);
 
         // Get operands
         Value* result = pass.buildOutputDefs(op, rewriter)[0];
         NGRAPH_CHECK(result, "Unexpected null result in Convolution Op");
         Value* images = operands[0];
         Value* filters = operands[1];
-        auto strides = convolOp.strides().getValue();
-        auto padBelow = convolOp.padBelow().getValue();
-        auto padAbove = convolOp.padBelow().getValue();
+        auto strides = convolOp.strides();
+        auto padBelow = convolOp.padBelow();
+        auto padAbove = convolOp.padBelow();
+
+        lowerConvolution(result,
+                         images,
+                         filters,
+                         strides,
+                         padBelow,
+                         padAbove,
+                         rewriter,
+                         pass,
+                         convolOp.getLoc());
 
-        Type elemTy = images->getType().cast<MemRefType>().getElementType();
+        rewriter.replaceOp(op, {result});
+        return matchSuccess();
+    }
+
+    REWRITER(NGGroupConvOp)
+    {
+        auto gConvOp = cast<NGGroupConvOp>(op);
+        ScopedContext scope(rewriter, gConvOp.getLoc());
+        // Get operands
+        Value* result = pass.buildOutputDefs(op, rewriter)[0];
+        NGRAPH_CHECK(result, "Unexpected null result in Convolution Op");
+        Value* images = operands[0];
+        Value* filters = operands[1];
+        auto strides = gConvOp.strides();
+        auto padBelow = gConvOp.padBelow();
+        auto padAbove = gConvOp.padBelow();
+        int groups = gConvOp.groups().getSExtValue();
+
+        NGRAPH_CHECK(groups > 0, "Invalid number of groups");
+        // create outer group convolution loop
+        // for group = 0 to groups
+        IndexHandle iv;
+
+        ValueHandle lb = intrinsics::constant_index(0);
+        ValueHandle ub = intrinsics::constant_index(groups);
+        ValueHandle step = intrinsics::constant_index(1);
+
+        auto imagesType = images->getType().cast<MemRefType>();
+        auto filtersType = filters->getType().cast<MemRefType>();
+        auto imagesShape = imagesType.getShape();
+        auto filtersShape = filtersType.getShape();
+
+        // Filters shape contains num of groups ?
+        bool groupsInFilters = (filtersShape.size() != imagesShape.size());
```

Comment:
is this check ok when group = 1?

Reply:
I think so. This flag is just to tell the rest of the code how to parse the filter dims. If the filter rank is reduced due to a leading dim=1 (groups), then the rest of the code will treat it as C_OUT, CHW.